CMakeLists.txt

```diff
@@ -163,7 +163,11 @@ include(cmake/modules/contrib/NNPack.cmake)
 
 add_library(tvm SHARED ${COMPILER_SRCS} ${RUNTIME_SRCS})
 add_library(tvm_topi SHARED ${TOPI_SRCS})
-add_library(tvm_runtime SHARED ${RUNTIME_SRCS})
+if (USE_CUDA AND MSVC)
+  add_library(tvm_runtime SHARED ${HALIDEIR_SRCS} ${RUNTIME_SRCS})
```

Comment:
runtime should never link HalideIR_SRCs, if there is anything wrong in link error, please report what is the specific error

Reply:
@tqchen thank you for the comment.
I got following linker errors without adding `HALIDEIR_SRCS` when I built tvm with use_cuda on Visual Studio 2017.

## cmake command

```
cmake -G "Visual Studio 15 2017 Win64" -DCMAKE_BUILD_TYPE=Release -DCMAKE_CONFIGURATION_TYPES="Rel
ease" ..
```

## error message

Severity | Code | Description | Project | File | Line | Suppression State
-- | -- | -- | -- | -- | -- | --
Error | LNK1120 | 20 unresolved externals | tvm_runtime | C:\work\tvm\build\Release\tvm_runtime.dll | 1 |  
Error | LNK2019 | unresolved external symbol "public: bool __cdecl HalideIR::float16_t::operator<(struct HalideIR::float16_t)const " (??Mfloat16_t@HalideIR@@QEBA_NU01@@Z) referenced in function "public: bool __cdecl HalideIR::float16_t::operator<=(struct HalideIR::float16_t)const " (??Nfloat16_t@HalideIR@@QEBA_NU01@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2019 | unresolved external symbol "public: bool __cdecl HalideIR::float16_t::operator==(struct HalideIR::float16_t)const " (??8float16_t@HalideIR@@QEBA_NU01@@Z) referenced in function "public: bool __cdecl HalideIR::float16_t::operator!=(struct HalideIR::float16_t)const " (??9float16_t@HalideIR@@QEBA_NU01@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2019 | unresolved external symbol "public: bool __cdecl HalideIR::float16_t::operator>(struct HalideIR::float16_t)const " (??Ofloat16_t@HalideIR@@QEBA_NU01@@Z) referenced in function "public: bool __cdecl HalideIR::float16_t::operator>=(struct HalideIR::float16_t)const " (??Pfloat16_t@HalideIR@@QEBA_NU01@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2019 | unresolved external symbol "public: bool __cdecl HalideIR::Type::same_handle_type(struct HalideIR::Type const &)const " (?same_handle_type@Type@HalideIR@@QEBA_NAEBU12@@Z) referenced in function "public: static struct HalideIR::Expr __cdecl HalideIR::Internal::BinaryOpNode<struct HalideIR::Internal::Add>::make(struct HalideIR::Expr,struct HalideIR::Expr)" (?make@?$BinaryOpNode@UAdd@Internal@HalideIR@@@Internal@HalideIR@@SA?AUExpr@3@U43@0@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2019 | unresolved external symbol "public: static unsigned int __cdecl tvm::Node::TypeKey2Index(char const *)" (?TypeKey2Index@Node@tvm@@SAIPEBD@Z) referenced in function "public: virtual bool const __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::Add>::_DerivedFrom(unsigned int)const " (?_DerivedFrom@?$ExprNode@UAdd@Internal@HalideIR@@@Internal@HalideIR@@UEBA?B_NI@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2019 | unresolved external symbol "public: virtual bool const __cdecl tvm::Node::_DerivedFrom(unsigned int)const " (?_DerivedFrom@Node@tvm@@UEBA?B_NI@Z) referenced in function "public: virtual bool const __cdecl HalideIR::Internal::BaseExprNode::_DerivedFrom(unsigned int)const " (?_DerivedFrom@BaseExprNode@Internal@HalideIR@@UEBA?B_NI@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::Add>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UAdd@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::Div>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UDiv@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::EQ>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UEQ@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::GE>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UGE@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::GT>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UGT@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::LE>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@ULE@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::LT>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@ULT@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::Max>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UMax@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::Min>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UMin@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::Mod>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UMod@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::Mul>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UMul@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::NE>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@UNE@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2001 | unresolved external symbol "public: virtual void __cdecl HalideIR::Internal::ExprNode<struct HalideIR::Internal::Sub>::accept(class HalideIR::Internal::IRVisitor *,struct HalideIR::Expr const &)const " (?accept@?$ExprNode@USub@Internal@HalideIR@@@Internal@HalideIR@@UEBAXPEAVIRVisitor@23@AEBUExpr@3@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  
Error | LNK2019 | unresolved external symbol "public: __cdecl HalideIR::VarExpr::VarExpr(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,struct HalideIR::Type)" (??0VarExpr@HalideIR@@QEAA@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@UType@1@@Z) referenced in function "public: __cdecl tvm::Var::Var(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,struct HalideIR::Type)" (??0Var@tvm@@QEAA@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@UType@HalideIR@@@Z) | tvm_runtime | C:\work\tvm\build\cuda_device_api.obj | 1 |  